### ISSUE TITLE
Update twisted/plugins to Py3 syntax enough to run trial

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#7915]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)| @adiroiban reported|
|-|-|
|Trac ID|trac#7915|
|Type|enhancement|
|Created|2015-05-26 15:45:52Z|
|Branch|https://github.com/twisted/twisted/tree/twisted-plugins-py3-7915-2|

I don't know if trial actually make use of the twisted/plugins but they are loaded by trial

The errors are simple

```
  File "/home/adi/chevah/twisted/build/py33/lib/python3.3/site-packages/zope/interface/declarations.py", line 412, in implements
    raise TypeError(_ADVICE_ERROR % 'implementer')
builtins.TypeError: Class advice impossible in Python3.  Use the @implementer class decorator instead.
```

Attachments:

* [foo](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/c2c/c2c44c89c9c9865055294af1468c84dff7e143e1/0beec7b5ea3f0fdbc95d0dd47f3c5bc275da8a33) (180 bytes) - added by moshez on 2015-05-31 23:54:10Z - Example for adi

<details><summary>Searchable metadata</summary>

```
trac-id__7915 7915
type__enhancement enhancement
reporter__adiroiban adiroiban
priority__normal normal
milestone__Python_3_x Python-3.x
branch__branches_twisted_plugins_py3_7915_2 branches/twisted-plugins-py3-7915-2
branch_author__adiroiban adiroiban
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1432655152461950 1432655152461950
changetime__1434092734154802 1434092734154802
version__None None
owner__adiroiban adiroiban

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_1"></a>|@adiroiban commented|
|-|-|

(In [44961]) Branching to twisted-plugins-py3-7915.

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_2"></a>|@adiroiban commented|
|-|-|

(In [44978]) Branching to twisted-plugins-py3-7915-2.

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_3"></a>|@adiroiban removed owner|
|-|-|


# Problem


When I try to run `./build/py33/bin/python3 bin/trial twisted/web -x` I get errors from twisted/plugins due to py3 syntax error 


# Changes


I have updated the code from plugins loaded by trial to py3 syntax, together with other code loaded by those plugins.

The code is not ported in py3... but it can be imported in py3 without errors.

I am doing this to help bootstrap the trial py3 port.

No file was added to py3 dist file


# How to test


Run `python3 bin/trial twisted/web -x` and you should no longer see error related to code from twisted.plugins.

There are still errors... but the errors are in trial code and they are already fixed by Hawkie in a branch dedicated to trial

Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="moshez's avatar" src="https://avatars.githubusercontent.com/u/422703?s=50" width="50" height="50">](https://github.com/moshez)<a name="note_4"></a>|@moshez set owner to @adiroiban|
|-|-|

Hi Adi,

Thanks for working on this! I'm so excited to have trial work on Py3 :)

This needs to be fixed:
1. The topfile seems to be missing content?

You might also consider:

2. In twisted/conch/ssh/keys.py, "Util.number.bytes_to_long(signature[:20])," -- Can we add parens instead of violating DRY?

Please address and resubmit.

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_5"></a>|@adiroiban set owner to @moshez|
|-|-|

Thanks for the review.

For topfile, what content do you expect? :) This is just a refactoring, it does not add no new feature. The code is not ported to py3, it is just updated to compile on py3.
This is why I put it as .misc. Content of misc tickets is ignored.

For Util.number.bytes_to_long(signature[:20]) how should I use the parens ? I am more than happy to reduce duplication.

Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="moshez's avatar" src="https://avatars.githubusercontent.com/u/422703?s=50" width="50" height="50">](https://github.com/moshez)<a name="note_6"></a>|@moshez set owner to @adiroiban|
|-|-|

Hi Adi,

Thanks for your clarifications.

HawkOwl said that .misc do not need content, so my bad.
Anyway, as far as refactoring I attached an example.

Please either merge, or see if my example shows how to reduce code dup, and integrate + merge.

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_7"></a>|@adiroiban commented|
|-|-|

Thanks. Don't worry! I am also not 100% familiar with the review process.

I am very happy to see you doing the review :)

Note that the splice is not duplicated, only the call for conversion is duplicate.

The calls are:

```

list[:N]
list[N:]

```

-----

Thanks for not letting this go.

I did a poor job in the previous path.

I will updated it to

```
            numbers = [
                Util.number.bytes_to_long(n) for n in
                    (signature[:20], signature[20:])
                ]
```

Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_8"></a>|@adiroiban set status to `closed`|
|-|-|

(In [44988]) Merge twisted-plugins-py3-7915-2: Update twisted/plugins and imported code to py3 syntax to help port trial.

Author: adiroiban
Reviewer: moshez
Fixes: [#7915](https://github.com/twisted/twisted/issues/7915)

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_9"></a>|@adiroiban commented|
|-|-|

This also fixes [#6918](https://github.com/twisted/twisted/issues/6918)
